lib/statistics/statistical_test/chi_squared_test.rb

```diff
@@ -37,6 +37,59 @@ def self.goodness_of_fit(alpha, expected, observed)
           alternative: p_value <= alpha,
           confidence_level: 1 - alpha }
       end
+
+      # The following three functions serve to calculate a test of independence for contingency
+      # tables of the type
+      #
+      #      A   B
+      # X   20  18
+      # Y    7  35
+      #
+      # They have been tested using 2x2 and 3x3 tables.
+      #
+      def self.test_of_independence(alpha, observed)
+        expected = calculate_expected(observed)
+        df = (observed.size - 1) * (observed.transpose.size - 1)
+        chi_score = calculate_chi_score(observed, expected)
+        probability = 1.0 - Statistics::Distribution::ChiSquared.new(df).cumulative_function(chi_score)
+        p_value = 1.0 - probability
+    
+        {
+          chi_score: chi_score,
+          df: df,
+          probability: probability,
+          p_value: p_value,
+          alpha: alpha,
+          #null: alpha < p_value,
+          #alternative: p_value <= alpha,
```

Comment:
Let's keep this two entries. In the end, that's what we want to achieve, no? test against a null hypothesis.

Reply:
I meant to uncomment this two lines, so you can get it in the returned value 😀

Reply:
sorry, forgot to reply to this - have done this already :)